### PR TITLE
Various fixes

### PR DESCRIPTION
**Pull request by stefanceriu**
* Fix information leaking in RoomSummaryDetail logs
* Prevent crash popups when force quitting the application
* Cleanup crash detection popup display
* Wait for logout confirmation before changing the app state

### CODE REVIEW

File: ElementX/Sources/Application/AppCoordinator.swift [hunk 1]
```diff
@@ -255,21 +260,22 @@ class AppCoordinator: AppCoordinatorProtocol {
...
+            
+            //  regardless of the result, clear user data
+            userSessionStore.logout(userSession: userSession)
+            userSession = nil
+            notificationManager?.delegate = nil
+            notificationManager = nil
+            
+            stateMachine.processEvent(.completedSigningOut)
+            
+            hideLoadingIndicator()
```
Comment (pixlwave) [inline ElementX/Sources/Application/AppCoordinator.swift:277]: In theory we should still handle `isSoftLogout` separately to this path as you don't want to clear the user data. We should remember their user ID so all they need to enter is their password to continue from where they left off.

**Review by pixlwave — COMMENTED** [review]
One comment on soft logout, otherwise looks good to me.

File: ElementX/Sources/Application/AppCoordinator.swift (hunk 1 above)
Comment (stefanceriu) [reply ElementX/Sources/Application/AppCoordinator.swift:277]: Oh I'm so sorry, I had a guard in there that I managed to delete somehow

**Review by pixlwave — APPROVED** [review]
Ahhh, that makes sense. LGTM 👍

**Pull request merged**